bigtop-deploy/puppet/modules/gpdb/manifests/init.pp

```diff
@@ -0,0 +1,440 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class gpdb {
+
+
+  class deploy($roles) {
+    if ("gpdb-master" in $roles or "gpdb-segment" in $roles) {
+      include gpdb::common
+    }
+  }
+
+
+  class common($nodes, $gp_home, $db_base_dir, $master_db_port, $segment_db_port_prefix) {
+
+    include stdlib
+
+    class { 'gpdb::common::install_packages': }
+
+    class { 'gpdb::common::gpadmin_user':}
+
+    class { 'gpdb::common::prepare_file_structure':
+      base_dir => "$db_base_dir",
+      require  => Class['gpdb::common::gpadmin_user']
+    }
+
+    class { 'gpdb::common::prepare_db_dirs':
```

Comment:
An overall comment about the puppet code: it is one of the best practices to align "=>" to the same position within the structure. This way it is easier to make changes/read the code. Could you please reformat it accordingly. Sorry for not mentioning this earlier.